### PR TITLE
[G2M] Asana/vacays

### PR DESCRIPTION
**Pull request by woozyking**
vacays

with a minor adjustment to `formatDates()` util function

### CODE REVIEW

**Review by hyx131 — APPROVED** [review]

File: sources/asana/index.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  if (pre) {
+    post.content = `${s}\n${post.content}`
+  } else {
+    post.content += `\n${s}`
+  }
+  return post
+}
+
+if (require.main === module) {
+  this.getVacays({ after: '2021-02-08', before: '2021-02-12' }).then(JSON.stringify).then(console.log)
```
Comment (tlowande) [inline sources/asana/index.js:97]: what's the point of these dates `8-12 Feb` or are they Aug-Dec 😅 ?

File: daily.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 const { DateTime } = require('luxon')
 
-const {
-  sources: { github: { issuesByRange, reposByRange, enrichIssues, ignoreProjects, ignoreBotUsers, formatPreviously } },
-  targets: { slack: { uploadMD } },
-} = require('.')
+const { issuesByRange, reposByRange, enrichIssues, ignoreProjects, ignoreBotUsers, formatPreviously } = require('./sources/github')
+const { getVacays, formatVacays } = require('./sources/asana')
```
Comment (tlowande) [inline daily.js:4]: why adding these files to `index.js` if fetching from their original `dir`?

**Review by tlowande — APPROVED** [review]
interesting workaround for the rate limiter with `timeout`

File: sources/asana/index.js (hunk 1 above)
Comment (woozyking) [reply sources/asana/index.js:97]: ah yea this is test code that runs locally. the `if (require.main === module)` is only invoked when doing:

```js
% node source/asana/index.js
```

but not doing anything when:

```js
const asana = require('./sources/asana')
```

File: daily.js (hunk 2 above)
Comment (woozyking) [reply daily.js:4]: wanted to move them in backward compatible ways, with a remote chance of other ppl already using this as a library and used to importing things from root `.`.

**Comment by woozyking** [comment]
> interesting workaround for the rate limiter with `timeout`

yea this is how the year-long timesheet can be generated by https://github.com/EQWorks/effort

**Pull request merged**